### PR TITLE
Added new ways to search for customers

### PR DESCRIPTION
**Pull request by kashing555**
Add ways to search by:
1. Name
2. Phone Number
3. Skin Type
4. Hair Type

### CODE REVIEW

File: src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package seedu.trackbeau.model.customer;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.trackbeau.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Data} matches any of the keywords given.
```
Comment (hanhuiice) [inline src/main/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicate.java:10]: "Person" should be "Customer"

**Review by hanhuiice — COMMENTED** [review]
LGMT! Did not find any errors in implementation.

File: src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java [hunk 2]
```diff
@@ -1,33 +1,59 @@
...
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<searchAreaKeyWord>\\S+)(?<arguments>.*)");
+
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
+    public FindCommand parse(String userInput) throws ParseException {
```
Comment (isabelteo) [inline src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java:28]: maybe can consider shortening method by abstracting out the assignment of matcher and handling of trimmedArgs so the exceptions are thrown by other methods

File: src/test/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicateTest.java [hunk 3]
```diff
@@ -41,34 +44,37 @@ public void equals() {
...
         assertFalse(predicate.test(new CustomerBuilder().withName("Alice").build()));
 
         // Non-matching keyword
-        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Carol"));
+        predicate = new SearchContainsKeywordsPredicate("getName", Arrays.asList("Carol"));
         assertFalse(predicate.test(new CustomerBuilder().withName("Alice Bob").build()));
 
         // Keywords match phone, email and trackbeau, but does not match name
-        predicate = new NameContainsKeywordsPredicate(Arrays.asList("12345", "alice@email.com", "Main", "Street"));
+        predicate = new SearchContainsKeywordsPredicate("getName",
```
Comment (isabelteo) [inline src/test/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicateTest.java:76]: can consider adding tests for the other searches in another PR!

File: src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java [hunk 4]
```diff
@@ -1,33 +1,59 @@
...
-        String[] nameKeywords = trimmedArgs.split("\\s+");
-
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        List<String> argumentList = Arrays.asList(trimmedArgs.split("\\s+"));
+
+        switch (searchAreaKeyWord.toLowerCase()) {
+        case "name":
+            return new FindCommand(new SearchContainsKeywordsPredicate("getName", argumentList));
+        case "phone":
+            return new FindCommand(new SearchContainsKeywordsPredicate("getPhone", argumentList));
```
Comment (isabelteo) [inline src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java:50]: has the user guide been updated with the search commands?

File: src/test/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicateTest.java (hunk 3 above)
Comment (kashing555) [reply src/test/java/seedu/trackbeau/model/customer/SearchContainsKeywordsPredicateTest.java:76]: yup will do that adding additional searches as well

File: src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java (hunk 4 above)
Comment (kashing555) [reply src/main/java/seedu/trackbeau/logic/parser/FindCommandParser.java:50]: nope will be added on the next PR with additional searches

**Review by flairekq — COMMENTED** [review]
LGTM!

**Pull request merged**